app/service/rest.py

```diff
@@ -262,10 +264,11 @@ def update_service(service_id):
     if 'letter_branding' in req_json:
         letter_branding_id = req_json['letter_branding']
         service.letter_branding = None if not letter_branding_id else LetterBranding.query.get(letter_branding_id)
-    dao_update_service(service)
 
-    if 'default_email_is_french' in req_json:
-        service.default_email_is_french = req_json['default_email_is_french']
+    if 'message_limit' in req_json:
+        redis_store.delete(daily_limit_cache_key(service_id))
```

Comment:
will there be any time where the value is the same and now what's happening is the cache is getting deleted for no reason?
i.e. should we check the new value of `message_limit` for equality before deleting the cache?

Reply:
Good question! The [admin code](https://github.com/cds-snc/notification-admin/blob/master/app/notify_client/service_api_client.py) passes the `message_limit` explicitly when updating a service so it seems safe to assume that if this key is here, the value has changed.

Updating a service does not happen often no matter what. It's done from service settings by service owners or a platform admin

Reply:
Improved the logic in the next commit